dask/array/tests/test_reductions.py

```diff
@@ -725,6 +725,33 @@ def test_object_reduction(method):
     assert result == 1
 
 
+@pytest.mark.parametrize("func", ["nanmin", "nanmax"])
+def test_empty_chunk_nanmin_nanmax(func):
+    # see https://github.com/dask/dask/issues/8352
+    x = np.arange(10).reshape(2, 5)
+    d = da.from_array(x, chunks=2)
+    x = x[x > 4]
+    d = d[d > 4]
+    block_lens = np.array([len(x.compute()) for x in d.blocks])
+    assert 0 in block_lens
+    assert_eq(getattr(da, func)(d), getattr(np, func)(x))
+
+
+@pytest.mark.parametrize("func", ["nanmin", "nanmax"])
+def test_empty_chunk_nanmin_nanmax_raise(func):
+    # see https://github.com/dask/dask/issues/8352
+    x = np.arange(10).reshape(2, 5)
+    d = da.from_array(x, chunks=2)
+    d = d[d > 9]
+    x = x[x > 9]
+    with pytest.raises(ValueError) as err_np:
+        x = getattr(np, func)(x)
```

Comment:
This is a nitpick but since this `x` isn't used and doesn't actually get set, we don't need it:

```suggestion
        getattr(np, func)(x)
```